floss/features/extract.py

```diff
@@ -97,13 +111,58 @@ def extract_function_calls_to(f):
     yield CallsTo(f.vw, [x[0] for x in f.vw.getXrefsTo(f.va, rtype=vivisect.const.REF_CODE)])
 
 
+def extract_function_kinda_tight_loop(f):
+    """
+    Yields tight loop features in the provided function
+    Algorithm by Blaine S.
+    """
+    for bb in f.basic_blocks:
+        # skip first and last BBs
+        if bb.va == f.va or bb.va + bb.size == f.basic_blocks[-1].va:
+            continue
+
+        # skip blocks that don't have exactly 2 successors
+        succs = list(f.get_bb_succs(bb))
+        if len(succs) != 2:
+            continue
+
+        # find the intermediate "loop" block that makes this "kinda" tight
+        loop_bb = None
+        for suc in succs:
+            if bb.va in [s.va for s in f.get_bb_succs(suc)]:
+                # successor must only have one successor itself
+                # successor can be the block itself (that's fine)
+                # without these rules it leaves room for weird corner cases
+                if len(list(f.get_bb_succs(suc))) == 1 or bb.va == suc.va:
```

Comment:
we should save off `f.get_bb_succs(suc)` in a temporary variable here to reduce the number of times this is computed